### PR TITLE
test: component test updates

### PR DESCRIPTION
**Pull request by marktnoonan**
- Closes multiple component testing stories from https://cypress-io.atlassian.net/browse/UNIFY-759

### Additional details

In the order GH presents the files, this PR:

- uses a label to locate the 'spec' input the generator in e2e tests
- uses label to locate input in SelectCloudProjectModal test
- adds a little more RecordKey test coverage
- uses a label to locate the 'spec' input in component tests
- adds a label to the Specs List search input
- adds a label to the empty generator input
- commits missing change for e2eProjectDirs 
- adds more testing to Alert component
- fixes Alert props - `alertClass` is confusing and targets the same element as the header, so it wasn't having any effect when used as the tests suggest it should be used.
- adds coverage to Input component. Would like to make Input also manage the `label` but for this PR I sidestepped that and fixed labels in consumers instead. The style of labels varies a lot.
- Unskip and fix a BaseError test for the retry button

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/app/src/specs/SpecsList.cy.tsx
Comment (lmiller1990) [inline packages/app/src/specs/SpecsList.cy.tsx:44]: No need to answer or change anything here right now, but are we sticking w/ i18n in specs or not? there's a ton of discussion around this, it's not entirely clear what we landed on.

File: packages/app/src/specs/SpecsList.cy.tsx
Comment (lmiller1990) [inline packages/app/src/specs/SpecsList.cy.tsx:47]: Not your change but what does `delay: 0` do here? (why do we need it)

**Review by lmiller1990 — DISMISSED** [review]

File: packages/app/src/specs/SpecsList.cy.tsx
Comment (marktnoonan) [reply packages/app/src/specs/SpecsList.cy.tsx:44]: Last I recall we were "sticking with it" as a practice but not to the point where we would hold up a PR that doesn't do it.

File: packages/app/src/specs/SpecsList.cy.tsx
Comment (marktnoonan) [reply packages/app/src/specs/SpecsList.cy.tsx:47]: I think it's just cumulatively saving some time across tests (vs the 10ms default)? We do it several places, @JessicaSachs might know if there's a bigger reason.

**Review by elevatebart — APPROVED** [review]
Reapproved after dismissing @lmiller1990 approval

**Pull request merged**
